atom/browser/native_window_views_win.cc

```diff
@@ -299,6 +300,68 @@ bool NativeWindowViews::PreHandleMSG(UINT message,
   }
 }
 
+void NativeWindowViews::HandleSizingEvent(WPARAM w_param, LPARAM l_param) {
+  double aspect_ratio = GetAspectRatio();
+  // aspect_ratio width/height
+  if (aspect_ratio == 0) {
+    return;
+  }
+
+  RECT* window_rect;
+  window_rect = reinterpret_cast<RECT*>(l_param);
+  int width = window_rect->right - window_rect->left;
+  int height = window_rect->bottom - window_rect->top;
+
+  double result_width = 0;
+  double result_height = 0;
+
+  double temp_width = 0;
+  double temp_height = 0;
+
+  gfx::Size ratio_fit_size(width, height);
+  gfx::Size min_fit_size = GetMinimumSize();
+
+  if (min_fit_size.width() >= width) {
+    width = min_fit_size.width();
+  } else if (min_fit_size.height() >= height) {
+    height = min_fit_size.height();
+  }
+
+  switch ((UINT)w_param) {
+    case WMSZ_LEFT:
+    case WMSZ_RIGHT:
+      result_width = width;
+      result_height = static_cast<double>(width) / aspect_ratio;
+      break;
+    case WMSZ_TOP:
+    case WMSZ_BOTTOM:
+      result_width = static_cast<double>(height) * aspect_ratio;
+      result_height = height;
+      break;
+    case WMSZ_TOPLEFT:
+    case WMSZ_TOPRIGHT:
+    case WMSZ_BOTTOMLEFT:
+    case WMSZ_BOTTOMRIGHT:
```

Comment:
This is the only place where `temp_width` and `temp_height` is used, so it might be slightly cleaner to wrap this case block in braces and move the temps into that scope.